en/computer_vision/path_planning_interface.md

```diff
@@ -0,0 +1,99 @@
+- Point 1 - Current waypoint (Unmodified/not type adapted)):
+  - `pos_x[1]`, `pos_y[1]`, `pos_z[1]`: x-y-z NED local position of *current* mission waypoint
+  - `vel_x[1]`, `vel_y[1]`, `vel_z[1]`: NaN
+  - `acc_x[1]`, `acc_y[1]`, `acc_z[1]`: NaN
+  - `pos_yaw[1]`: Yaw setpoint
+  - `vel_yaw[1]`: Yaw speed setpoint
+  - `command[1]`: The [MAVLink Command](https://mavlink.io/en/messages/common.html#mavlink-commands-mavcmd) for the current waypoint.
+- Point 2 - Next waypoint in local coordinates (unmodified/not type adapted):
+  - `pos_x[2]`, `pos_y[2]`, `pos_z[2]`: x-y-z NED local position of *next* mission waypoint
+  - `vel_x[2]`, `vel_y[2]`, `vel_z[2]`: NaN
+  - `acc_x[2]`, `acc_y[2]`, `acc_z[2]`: NaN
+  - `pos_yaw[2]`: Yaw setpoint
+  - `vel_yaw[2]`: Yaw speed setpoint
+  - `command[2]`: The [MAVLink Command](https://mavlink.io/en/messages/common.html#mavlink-commands-mavcmd) for the next waypoint.
+- All other indices/fields are set as NaN.
+
+<span id="type_adapted"></span>
+Notes:
+- Point 0 is the current waypoint/target modified based on the type of target.
+  For example, it makes sense when landing to specify the target x, y coordinates and a descent velocity.
+  To achieve this `FlightTaskAutoMapper` modifies land waypoints in Point 0 to set the z component of position to NAN and the z-velocity to a desired value.
+- Point 1 and 2 are not used by the safe landing planner.
+- Point 1 is used by local and global planner.
+
+On the companion side, MAVROS translates the MAVLink message into ROS messages, which are eventually handled by the appropriate planner.
+The planner plans a path to the waypoint/target, and sends it to the vehicle as a stream of `TRAJECTORY_REPRESENTATION_WAYPOINTS` messages that have the setpoint in Point 0.
+
+The fields for the `TRAJECTORY_REPRESENTATION_WAYPOINTS` message from the companion computer are set as shown:
+- `time_usec`: UNIX Epoch time.
+- `valid_points`: 1
+- Current vehicle information:
+  - `pos_x[0]`, `pos_y[0]`, `pos_z[0]`: x-y-z NED vehicle local position setpoint
+  - `vel_x[0]`, `vel_y[0]`, `vel_z[0]`: x-y-z NED velocity setpoint
+  - `acc_x[0]`, `acc_y[0]`, `acc_z[0]`: NaN
+  - `pos_yaw[0]`: Yaw angle setpoint
+  - `vel_yaw[0]`: Yaw speed setpoint
+  - `command[0]`: NaN.
+- All other indices/fields are set as NaN.
+
+
+
+## Companion Failure Handling
+
+PX4 safely handles the case where messages are not received from the offboard system:
+- If the `HEARTBEAT` is lost, PX4 will emit a status message (which is displayed in QGroundControl) stating either "Avoidance system lost" or "Avoidance system timeout" (depending on the vehicle state).
+- When external path planning is enabled (`COM_OBS_AVOID=1`) and the vehicle is in an autonomous mode (Hold, Return, Mission, Takeoff, Land) the vehicle will switch into [Hold mode](../flight_modes/hold.md) a trajectory message is not received for more than 0.5 seconds.
```

Comment:
OK, @mrivi Info on failure from https://github.com/PX4/px4_user_guide/pull/510/#discussion_r301443267 now added to this section.

1. This text slightly different than what you said - basically you talked about messages being old, but I have made it like "if message not received".
2. The assumption therefore is that the planner will always be able to calculate the next point within .5 second (so there is never a gap). Is this reasonable? Or does the planner send "old points" if it doesn't have new ones yet to ensure that it is always sending something?
3. So what if you enable avoidance but no planner is running? Does the same thing happen - just drop into hold immediately?
4. What if you have safe landing planner running, but you go into a mission. PX4 expects setpoints because it is in a mission but the safe landing planner isn't going to provide them until you get to a land mission item.  Won't this push you into hold mode?

Reply:
2. On the planner side we use a lightweight algorithm to give just the next point on the expensively-computed path, this should always be available in 0.5s.
3. Only if you switch to an AUTO mode
4. The Safe Landing system gives "I don't care what you do" setpoints (NaN setpoints), which lets the internal FCU setpoints continue with the mission undisturbed.

Reply:
> So what if you enable avoidance but no planner is running? Does the same thing happen - just drop into hold immediately?

If COM_OBS_AVOID is enabled from boot and no planner is running, the preflight checks will fails no matter in which mode the vehicle is. So you won't be able to fly until COM_OBS_AVOID is set to 0